gravitational/webapps
# [Discover] Add resource discovery polling and static OS logins

#### Description
part of https://github.com/gravitational/teleport/issues/13997

- Enabled flow (albeit hackily) for the sake of testing the flow
- Added polling during download script step, which just periodically queries for the resource given a search filter for `reference resource ID` which is returned by creation of a join token, and will be used by the script to uniquely label the added resource. 
- Changed `principal` language with `OS usernames` (can't remember if it was `OS usernames` or `OS logins`)
- Added the static logins that we retrieved by querying for the resource (currently fetching the user traits only returns the dynamic logins defined by the user). Since static logins cannot be modified it is disabled in the list.

#### Required before merge
- [ ] https://github.com/gravitational/teleport/pull/15114

#### How to test
1) You need an account from this cluster (DM me, I can't disable 2FA for cloud): https://kimlisa2.cloud.gravitational.io/web, this cluster contains @marcoandredinis backend changes
2) Check this branch out and in `packages/teleport/src/config.ts`, set the flag value `enabledDiscoverWizard` to `true`
3) Start dev server `yarn start-teleport-e --target=https://kimlisa2.cloud.gravitational.io/web`
4) Click on `Discovery Wizard` from dashboard top right corner menu and go through the flow.

Notes on testing:
- When you run the download script, you'll need to add `-k` flag to the curl command to use insecure connection
- You'll need to nuke any existing teleport data (the script will tell you how)

#### Demo

I'm kinda uncertain about the language i used for the different polling states, any suggestions to make it clearer to user would be great. I'm also uncertain what to do for the error state. Currently it provides a user a button to regenerate script and start polling again. @xinding33 

happy path demo (installer takes some time):

![polling-demo2](https://user-images.githubusercontent.com/43280172/183020010-bd56512f-5505-4081-a57b-6439f00491cc.gif)

polling state:

![image](https://user-images.githubusercontent.com/43280172/183020338-da4685d1-6a48-4a51-8c08-df801d7d3869.png)

success state:

![image](https://user-images.githubusercontent.com/43280172/183012391-df260fb3-da7b-452e-8ba5-ca299939d92c.png)

error state:

![image](https://user-images.githubusercontent.com/43280172/183020451-3cf55265-8699-47bc-b18f-6e738f9dc415.png)
